src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/SemanticTokens.cs

```diff
@@ -85,10 +86,12 @@ internal partial class RazorCustomMessageTarget
         var nonEmptyResults = results.Select(r => r?.Response?.Data).WithoutNull().ToArray();
         if (nonEmptyResults.Length != semanticTokensParams.Ranges.Length)
         {
+            _logger?.LogDebug("Made {count} semantic tokens requests to Roslyn but only got {nonEmpty} results back", semanticTokensParams.Ranges.Length, nonEmptyResults.Length);
             // Weren't able to re-invoke C# semantic tokens but we have to indicate it's due to out of sync by providing the old version
             return new ProvideSemanticTokensResponse(tokens: null, hostDocumentSyncVersion: csharpDoc.HostDocumentSyncVersion ?? -1);
         }
 
+        _logger?.LogDebug("Made {count} semantic tokens requests to Roslyn", semanticTokensParams.Ranges.Length);
```

Comment:
Snuck this one in too, so we can see what is going on with our new feature flag.